data/pigui/modules/info-view/06-flightlog.lua

```diff
@@ -37,191 +59,170 @@ local function headerText(title, text, wrap)
 	end
 end
 
+function ui_formatter:write( string )
+	ui.text( string )
+	ui.sameLine()
+	return self
+end
 
-local entering_text_custom = false
-local entering_text_system = false
-local entering_text_station = false
+function ui_formatter:newline()
+	ui.text( "" )
+	return self
+end
+
+local entering_text = false
 
 -- Display Entry text, and Edit button, to update flightlog
-local function inputText(entry, counter, entering_text, log, str, clicked)
-	if #entry > 0 then
-		headerText(l.ENTRY, entry, true)
-	end
+local function inputText(entry, counter, entering_text, str)
 
-	if clicked or entering_text == counter then
+	if entering_text == counter then
 		ui.spacing()
 		ui.pushItemWidth(-1.0)
-		local updated_entry, return_pressed = ui.inputText("##" ..str..counter, entry, {"EnterReturnsTrue"})
+		local updated_entry, return_pressed = ui.inputText("##" ..str..counter, entry:GetEntry(), {"EnterReturnsTrue"})
 		ui.popItemWidth()
-		entering_text = counter
 		if return_pressed then
-			log(counter, updated_entry)
+			entry:UpdateEntry(updated_entry)
 			entering_text = -1
 		end
+	elseif entry:HasEntry() then
+		ui_formatter:headerText(l.ENTRY, entry:GetEntry(), true)
 	end
 
 	ui.spacing()
 	return entering_text
 end
 
--- Based on flight state, compose a reasonable string for location
-local function composeLocationString(location)
-	return string.interp(l["FLIGHTLOG_"..location[1]],
-		{ primary_info = location[2],
-			secondary_info = location[3] or "",
-			tertiary_info = location[4] or "",})
-end
+local function renderLog( formatter )
 
-local function renderCustomLog()
-	local counter = 0
-	local was_clicked = false
-	for systemp, time, money, location, entry in FlightLog.GetCustomEntry() do
-		counter = counter + 1
-
-		headerText(l.DATE, formatDate(time))
-		headerText(l.LOCATION, composeLocationString(location))
-		headerText(l.IN_SYSTEM, ui.Format.SystemPath(systemp))
-		headerText(l.ALLEGIANCE, systemp:GetStarSystem().faction.name)
-		headerText(l.CASH, Format.Money(money))
-
-		::input::
-		entering_text_custom = inputText(entry, counter,
-			entering_text_custom, FlightLog.UpdateCustomEntry, "custom", was_clicked)
-		ui.nextColumn()
+	ui.spacing()
+	-- input field for custom log:
+	ui_formatter:headerText(l.LOG_NEW, "")
+	ui.sameLine()
+	local text, changed = ui.inputText("##inputfield", "", {"EnterReturnsTrue"})
+	if changed then
+		FlightLog.MakeCustomEntry(text)
+	end
+	ui.separator()
 
-		was_clicked = false
-		if ui.iconButton(icons.pencil, buttonSpaceSize, l.EDIT .. "##custom"..counter) then
-			was_clicked = true
-			-- If edit field was clicked, we want to edit _this_ iteration's field,
-			-- not next record's. Quick, behind you, velociraptor!
-			goto input
+	local counter = 0
+	for entry in FlightLog:GetLogEntries(getIncludedSet(), nil, earliest_first ) do
+	 	counter = counter + 1
+	
+		 writeLogEntry( entry, formatter, true )
+
+		 if entry:CanHaveEntry() then
+			::input::
```

Comment:
As I've discussed with you prior, we prefer not to use `goto` unless absolutely necessary. I don't see a surviving `goto` call in the diff - if that's been removed then I'd prefer we also remove the label as well.

Reply:
(I _think_ I was the original author to put in `goto` in this code. Thus might be a remnant from my first implementation of it... just to not put blame for such an atrocious act of violence upon poor @JonBooth78 ).

Reply:
Ahh, yes, sorry; I'd missed the label but have I successfully removed the `goto`